### PR TITLE
Feature/dot net3.5

### PR DESCRIPTION
**Pull request by njbmartin**
A few changes made in order to improve compatibility with Unity of course, general .NET 3.5 compatibility.

- Downgraded .NET Framework target to 3.5
- Implemented EnvironmentHelper from [StackOverflow answer](http://stackoverflow.com/a/28866330/1104531) to determine the OS architecture directly instead of from the CLR (Unity didn't like this)

Closes #45 and #41

### CODE REVIEW

File: Corale.Colore/EnvironmentHelper.cs [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+//
+//     "Razer" is a trademark of Razer USA Ltd.
+// </copyright>
+// ---------------------------------------------------------------------------------------
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Runtime.InteropServices;
+using System.Text;
```
Comment (Sharparam) [inline Corale.Colore/EnvironmentHelper.cs]: Using statements should be inside namespace.

File: Corale.Colore/EnvironmentHelper.cs [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+        static extern IntPtr GetCurrentProcess();
+
+        [DllImport("kernel32.dll")]
+        static extern IntPtr GetModuleHandle(string moduleName);
+
+        [DllImport("kernel32")]
+        static extern IntPtr GetProcAddress(IntPtr hModule, string procName);
+
+        [DllImport("kernel32.dll")]
+        static extern bool IsWow64Process(IntPtr hProcess, out bool wow64Process);
```
Comment (Sharparam) [inline Corale.Colore/EnvironmentHelper.cs]: Kernel32 interop should be in the Kernel32 NativeMethods file.

File: Corale.Colore/Razer/Result.cs [hunk 3]
```diff
@@ -347,10 +347,10 @@ public override string ToString()
             foreach (var fieldInfo in fieldsInfo.Where(fi => fi.FieldType == typeof(Result)))
             {
                 var value = fieldInfo.GetValue(null);
-                var attr = fieldInfo.GetCustomAttribute<DescriptionAttribute>(false);
+                var attr = fieldInfo.GetCustomAttributes(typeof(DescriptionAttribute),false)[0];
```
Comment (Sharparam) [inline Corale.Colore/Razer/Result.cs]: `GetCustomAttributes(Type, bool)` can return a 0-length array, so this should probably be `var attrs` instead, retrieving the whole array and then checking for `attr.Length > 0 && attr[0] is DescriptionAttribute)`.

Actually we wouldn't need to check the Type at all would we? If there is at least 1 element in the array, that means the DescriptionAttribute was found and it's restricted to 1 attribute per member anyway.

**Comment by njbmartin** [comment]
Fixing now

**Comment by brandonscott** [comment]
EnvironmentHelper.cs(35, 18): error CS0234: The type or namespace name 'Native' does not exist in the namespace 'Corale.Colore' (are you missing an assembly reference?)

File: Corale.Colore/Native/Kernel32/NativeMethods.cs [hunk 4]
```diff
@@ -52,7 +50,17 @@ internal static class NativeMethods
...
+
+        [DllImport(DllName, CharSet = CharSet.Ansi, EntryPoint = "GetModuleHandle", SetLastError = true,
+            ThrowOnUnmappableChar = true, BestFitMapping = false)]
+        internal static extern IntPtr GetModuleHandle(string moduleName);
+
+        [DllImport(DllName, CharSet = CharSet.Ansi, EntryPoint = "GetModuleHandle", SetLastError = true,
+            ThrowOnUnmappableChar = true, BestFitMapping = false)]
+        internal static extern bool IsWow64Process(IntPtr hProcess, out bool wow64Process);
+    }
+}
```
Comment (Sharparam) [inline Corale.Colore/Native/Kernel32/NativeMethods.cs]: Missing end of file newline

File: Corale.Colore/Razer/Result.cs [hunk 5]
```diff
@@ -214,7 +214,7 @@ public bool Success
         /// </summary>
         /// <param name="result">An instance of the <see cref="Result" /> to convert.</param>
         /// <returns>The integer equivalent of the <paramref name="result" />.</returns>
-        public static implicit operator int(Result result)
+        public static implicit operator int (Result result)
```
Comment (Sharparam) [inline Corale.Colore/Razer/Result.cs]: Remove this space

File: Corale.Colore/Razer/Result.cs [hunk 6]
```diff
@@ -237,7 +237,7 @@ public bool Success
         /// <c>true</c> if the <paramref name="result" /> represents a <c>true</c> value (success),
         /// otherwise <c>false</c> (failure).
         /// </returns>
-        public static implicit operator bool(Result result)
+        public static implicit operator bool (Result result)
```
Comment (Sharparam) [inline Corale.Colore/Razer/Result.cs]: Remove this space

File: Corale.Colore/EnvironmentHelper.cs [hunk 7]
```diff
@@ -0,0 +1,68 @@
...
+    {
+        /// <summary>
+        /// <returns>Returns true if the OS is 64-bit</returns>.
+        /// </summary>
+        public static bool Is64BitOperatingSystem()
+        {
+            // Check if this process is natively an x64 process. If it is, it will only run on x64 environments, thus, the environment must be x64.
+            if (IntPtr.Size == 8)
+                return true;
+            // Check if this process is an x86 process running on an x64 environment.
```
Comment (Sharparam) [inline Corale.Colore/EnvironmentHelper.cs]: SA1515 SingleLineCommentsMustBePrecededByBlankLine

File: Corale.Colore/EnvironmentHelper.cs [hunk 8]
```diff
@@ -0,0 +1,68 @@
...
+            {
+                IntPtr processAddress = NativeMethods.GetProcAddress(moduleHandle, "IsWow64Process");
+                if (processAddress != IntPtr.Zero)
+                {
+                    bool result;
+                    if (NativeMethods.IsWow64Process(NativeMethods.GetCurrentProcess(), out result) && result)
+                        return true;
+                }
+            }
+            // The environment must be an x86 environment.
```
Comment (Sharparam) [inline Corale.Colore/EnvironmentHelper.cs]: SA1515 SingleLineCommentsMustBePrecededByBlankLine

**Comment by Sharparam** [comment]
Looks good :+1:

**Pull request merged**